### PR TITLE
Add some class javadocs

### PR DESCRIPTION
**Pull request by SoniEx2**
They're not very good because I'm making them up cleanroom from the mappings alone.

### CODE REVIEW

File: mappings/net/minecraft/block/Block.mapping [hunk 1]
```diff
@@ -1,4 +1,5 @@
 CLASS net/minecraft/class_2248 net/minecraft/block/Block
+	COMMENT A block type.
```
Comment (i509VCB) [inline mappings/net/minecraft/block/Block.mapping:2]: I'd argue we don't really need javadoc in this capacity since this is a pretty universally known concept.

File: mappings/net/minecraft/item/Item.mapping [hunk 2]
```diff
@@ -1,4 +1,5 @@
 CLASS net/minecraft/class_1792 net/minecraft/item/Item
+	COMMENT An item type.
```
Comment (i509VCB) [inline mappings/net/minecraft/item/Item.mapping:2]: Similar justification to `Block`

File: mappings/net/minecraft/item/BlockItem.mapping [hunk 3]
```diff
@@ -1,4 +1,5 @@
 CLASS net/minecraft/class_1747 net/minecraft/item/BlockItem
+	COMMENT An item type for a block type.
```
Comment (i509VCB) [inline mappings/net/minecraft/item/BlockItem.mapping:2]: I would probably reword this to be a bit closer to the following:

```suggestion
	COMMENT Represents a block in item form.
	COMMENT This item handles the placement of a block in a world along with applying any block entity data present on the item stack.
```

**Review by i509VCB — CHANGES_REQUESTED** [review]

**Comment by SoniEx2** [comment]
We do believe javadocs for Block and Item are in order - Block could be an in-world location or subdivision (really, there's a free software voxel engine that does something like this), and Item could also be an in-world item or in-inventory item. And additionally, Java, Python, etc all have class docs for everything that's directly exposed to the API user.

If Block was some implementation detail, then sure, leave it without docs. but as far as we can tell it isn't.

**Comment by frqnny** [comment]
But with minecraft this is really unnecessary. Blocks are blocks and it'll be confusing if you use "block" for something else they'll look at you weird.

**Comment by SoniEx2** [comment]
Can't have yarnimpl without having these javadocs. :v

On a more serious note, we maintain our position.

File: mappings/net/minecraft/block/Block.mapping (hunk 1 above)
Comment (Gegy) [reply mappings/net/minecraft/block/Block.mapping:2]: If these do have documentation, it's probably worth mentioning that they are singleton instances. Especially for items, it's a common mistake to attach state to an item implementation which becomes unintentionally global.

Comment (SoniEx2) [reply mappings/net/minecraft/block/Block.mapping:2]: That's not what a singleton is.

**Comment by modmuss50** [comment]
The docs provided here dont provide any more help or context IMO. Ideally these classes would be fully documented but thats a large amount of work.
